src/internal/clusterstate/v2.8.0/pfsdb.go

```diff
@@ -130,3 +130,101 @@ func migrateRepos(ctx context.Context, tx *pachsql.Tx) error {
 	}
 	return nil
 }
+
+// alterCommitsTable1 adds useful new columns to pfs.commits table.
+// Note that this is not the end all be all. We will need to make more changes after data has been migrated.
+// TODO
+//   - rename int_id to id? This will requires changing all references as well.
+//   - make repo_id not null
+//   - make origin not null
+//   - make updated_at not null and default to current timestamp
+//   - branch_id_str is a metadata reference to the branches table. Today this points to collections, but tomorrow it should
+//     point to pfs.branches. Once the PFS master watches branches, we will no longer need this column.
+func alterCommitsTable1(ctx context.Context, tx *pachsql.Tx) error {
+	query := `
+	CREATE TYPE pfs.commit_origin AS ENUM ('ORIGIN_KIND_UNKNOWN', 'USER', 'AUTO', 'FSCK');
+
+	ALTER TABLE IF EXISTS pfs.commits
+	    DROP CONSTRAINT fk_col_commit,
+		ADD COLUMN IF NOT EXISTS repo_id bigint REFERENCES pfs.repos(id),
+		ADD COLUMN IF NOT EXISTS origin pfs.commit_origin,
+		ADD COLUMN IF NOT EXISTS description text DEFAULT '',
+		ADD COLUMN IF NOT EXISTS start_time timestamptz,
+		ADD COLUMN IF NOT EXISTS finishing_time timestamptz,
+		ADD COLUMN IF NOT EXISTS finished_time timestamptz,
+		ADD COLUMN IF NOT EXISTS compacting_time bigint,
+		ADD COLUMN IF NOT EXISTS validating_time bigint,
+		ADD COLUMN IF NOT EXISTS error text,
+		ADD COLUMN IF NOT EXISTS size bigint,
+		ADD COLUMN IF NOT EXISTS updated_at timestamptz DEFAULT CURRENT_TIMESTAMP,
+		ADD COLUMN IF NOT EXISTS branch_id_str text;
+
+	CREATE TRIGGER set_updated_at
+		BEFORE UPDATE ON pfs.commits
+		FOR EACH ROW EXECUTE PROCEDURE core.set_updated_at_to_now();
+	`
+	if _, err := tx.ExecContext(ctx, query); err != nil {
+		return errors.Wrap(err, "altering commits table")
+	}
+	return nil
+}
+
+// commitAncestry is how we model the commit graph within a single repo.
+func createCommitAncestryTable(ctx context.Context, tx *pachsql.Tx) error {
+	query := `
+	CREATE TABLE IF NOT EXISTS pfs.commit_ancestry (
+		from_id bigint REFERENCES pfs.commits(int_id),
+		to_id bigint REFERENCES pfs.commits(int_id),
+		PRIMARY KEY (from_id, to_id)
```

Comment:
This might not actually work. In Postgres, a primary key is both non-null and unique. However, for initial sentinel commits, the `to_id` is null. Instead of primary key, we should relaxed the non-null constraint and just apply unique constraint.

Reply:
This might be okay because we're only creating an entry here once we have a relationship. So the sentinel commit would not exist in this table until it pointed to a child commit.